src/Orleans/Core/Exceptions.cs

```diff
@@ -207,5 +207,29 @@ protected ClientNotAvailableException(SerializationInfo info, StreamingContext c
         { }
 #endif
     }
+
+    /// <summary>
+    /// Indicates that a <see cref="GrainReference"/> was not bound to the runtime before being used.
+    /// </summary>
+    [Serializable]
+    public class GrainReferenceNotBoundException : OrleansException
+    {
+        internal GrainReferenceNotBoundException(GrainReference grainReference) : base(CreateMessage(grainReference)) { }
+
+        private static string CreateMessage(GrainReference grainReference)
+        {
+            return $"Attempted to use a GrainReference which has not been bound to the runtime: {grainReference.ToDetailedString()}." +
+                   " Use the Bind method to bind this reference to the runtime.";
```

Comment:
I see that the `GrainReference.Bind` method is internal Maybe this should point to `{nameof(IGrainFactory)}.{nameof(BindGrainRereference)}` instead?